macros/src/function_like/println.rs

```diff
@@ -0,0 +1,40 @@
+use defmt_parser::ParserMode;
+use proc_macro::TokenStream;
+use proc_macro_error::abort;
+use quote::quote;
+use syn::parse_macro_input;
+
+use crate::construct;
+use crate::function_like::log::{Args, Codegen};
+
+pub(crate) fn expand(args: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(args as Args);
+
+    let format_string = args.format_string.value();
+    let fragments = match defmt_parser::parse(&format_string, ParserMode::Strict) {
+        Ok(args) => args,
+        Err(e) => abort!(args.format_string, "{}", e),
+    };
+
+    let formatting_exprs = args
+        .formatting_args
+        .map(|punctuated| punctuated.into_iter().collect())
+        .unwrap_or_else(Vec::new);
+
+    let Codegen { patterns, exprs } = Codegen::new(
+        &fragments,
+        formatting_exprs.len(),
+        args.format_string.span(),
+    );
+
+    let format_tag = construct::interned_string(&format_string, "println", false);
+    quote!({
+        match (#(&(#formatting_exprs)),*) {
+            (#(#patterns),*) => {
+                defmt::export::istr(&#format_tag);
```

Comment:
this should follow the acquire - release pattern used in [the log macros](https://github.com/knurling-rs/defmt/blob/main/macros/src/function_like/log.rs#L43-L46), including the `header` call. the main difference between the two will be that tag used to constructor the header (there's a `make_istr` call in the expansion I believe)